CONTRIBUTING.md

```diff
@@ -0,0 +1,90 @@
+changes, and helping you finalize your pull requests.
+
+As for everything else in the project, the contributions to Energy-Price-News-API are governed by our [Code of Conduct](https://github.com/Energy-Price-News-API/energy-prices-api/blob/master/CODE_OF_CONDUCT.md).
+
+### What kinds of contributions we are looking for.
+
+We keep an open mind! Improving documentation, bug triaging, or writing tutorials are all examples of helpful contributions for both the users of this tool **and** future contributors.
+
+## Your First Contribution
+
+The aim for this project it to be very easy to access, so please have a look through the [issues](https://github.com/Energy-Price-News-API/energy-prices-api/issues) listed to see if any suit your current or desired skills. The maintainers of the project will try to keep it all as up-to-date as possible.
+
+I would like to encourage contributions of any kind to this project:
+
+- Issues:
+  - Suggestions on how to do something
+  - Problems/bugs found in the code
+  - Ideas for features
+  - Discussion around best practices so that others can learn from your experience/knowledge
+- Pull Requests
+  - Should you have a feature you'd like to share already coded, please submit a detailed pull request
+  - A preferred PR format is not yet defined, so we ask you just use common sense and include the type of info you'd like to see as a maintainer
+- Documentation
+  - Spellchecking and typo corrections are **ALWAYS** welcome :joy:, seriously, I hate typos but make them frequently!
+  - Further detail. If you feel something needs to be expanded on, please do so and submit your PR.
+  - Wiki: If we get to a point that a wiki would be helpful, please open an issue/discussion with your suggestions
+  - As the project grows, this `CONTRIBUTING.md` file will need updating, so add to it where you see fit.
+- Share what you do
+  - In order to encourage others to join in, feel free to share what you have done on your social media/websites
+  - If you know someone that has done something similar, please show them this project as their input would be greatly appreciated
+
+### A link to a resource for people who have never contributed to open source before.
+Here are a couple of friendly tutorials you can check out: http://makeapullrequest.com/ and http://www.firsttimersonly.com/
+
+Working on your first Pull Request? You can learn how for *free* by checking [this repo, first-contributions](https://github.com/firstcontributions/first-contributions/blob/main/README.md).
+
+# Getting started
+### A quick walkthrough of how to submit a contribution.
+
+For something that is bigger than a one or two line fix:
+1. Create your own fork of the code
+2. Do the changes in your fork
+3. If you like the change and think the project could use it:
+    * Try to follow the code style for the project
+    * Note our Code of Conduct.
+    * Send a pull request.
+
+> **Note**
```

Comment:
I would suggest that you add the steps from forking to making a PR. 
Here are some examples: 
- https://docs.opensauced.pizza/contributing/introduction-to-contributing/
- https://github.com/EddieHubCommunity/LinkFree/blob/main/CONTRIBUTING.md
- https://github.com/TAKANOME-DEV/coffee-stores/blob/main/CONTRIBUTING.md
- https://docs.nhcarrigan.com/#/contributing

Besides that, everything looks great :+1:

Reply:
We should also follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for commit messages

Reply:
> information_source Suggestion We can follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) for commit messages. We can also use [commitlint](https://commitlint.js.org/) and [pre-commit hook](https://prettier.io/docs/en/precommit.html) for linting the commit message.

I'll look into that later cos I think the documentation may need whittling down a bit before adding more steps. Also I am starting to think a bit of a refactor may be on the cards regarding the controllers etc. But all in due time!

Reply:
The problems I remember facing are:

1. Ensuring correct version of Node.js
2. Running `git checkout master && git pull` before making a branch (particularly contributors that contribute a **second** time!)
3. Waiting to be assigned an issue

Those topics I think I could maybe cover better. But which parts do I drop?? :sweat_smile:

Reply:
I think we can force the version of Nodejs to use, have a look [here](https://github.com/blimmer/enforced-package-json-engines)

Reply:
I will open a new issue for forcing Node version with a reference to the above